libs/hwManager/communication.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This file is use for the exchange messages with the HWManager.
+ * The communication message is through IPC (window.ipc)
+ */
+import { IPC_MESSAGES } from './constants';
+
+const IPC = window.ipc;
+
+/**
+ * sendCommand - Function.
+ * Use for send and request data to the HWManager.
+ */
+const sendCommand = (action, payload) => {
+  // eslint-disable-next-line no-new
+  new Promise((resolve, reject) => {
+    // Listening for response
+    IPC.once(`${action}.response`, (event, response) => {
+      if (response.success) return resolve(response.data);
+      return reject(new Error(`${action} failed`));
+    });
+    // Requesting data
+    IPC.send(`${action}.request`, payload);
+  });
+};
+
+/**
+ * getPublicKey - Function.
+ * Use for get the public key from the device for a specific account(s)
+ * @param {object} data -> Object that contain the information about the device and data
+ * @param {string} data.deviceId -> Id of the hw device
+ * @param {number} data.index -> index of the account of wich will extact information
+ * @param {boolean} data.showOnDevice -> Boolean value to inform device if show or
+ * not information in screen
+ */
+const getPublicKey = async (data) => {
+  const response = await sendCommand(IPC_MESSAGES.GET_PUBLICK_KEY, data);
+  return response;
+};
+
+/**
+ * signTransaction - Function.
+ * Use for sign a transaction, this could be send or vote
+ * @param {object} data -> Object that contain the information about the device and data
+ * @param {string} data.deviceId -> Id of the hw device
+ * @param {number} data.index -> index of the account of wich will extact information
+ * @param {string} data.message -> Data to display in the device screen
+ * @param {boolean} data.showOnDevice -> Boolean value to inform device if show or
```

Comment:
The docs for properties of `data` look like copy-paste from `getPublicKey` and not for `signTransaction`.

Reply:
oh it is because the object is the same with the difference of the message that only apply for send/vote transactions

Reply:
`data.showOnDevice` doesn't make sense for `signTransaction`. There is no way to signa transaction without showing it on device.
YOU need these 3 in `data` object for `signTransaction`
https://github.com/LiskHQ/lisk-hub/blob/91395a47fa2ec5b19b464251bb9d203deb1f39e5/src/utils/api/hwWallet.js#L207-L209

The jsDoc comment is lying 😉

Reply:
yes, you are right for signTx we need to add tx param in the data object, but we dont need it for publicKey